### DISCUSSION TITLE
How can I get a merged result from model A and his sub-model B?

### DISCUSSION THREAD

**Discussion by Jasminecll** [sqlalchemy/sqlalchemy#8156] [Usage Questions]
If I have A model, like this:

    class A: 

        id = Column(Integer, primary_key=True)

        column_a = Column(String(32))

        column_b = Column(String(32)) 

and his sub-model B:

    class B: 

        id = Column(Integer, primary_key=True)

        class_A_id = Column(Integer, unique=True)

        column_c = Column(String(32))

I want to get a result include A and his relaionship column_c, what should I do?

I use Async sqlalchemy, and the version is 1.4.

I try to use:

      stmt = select(A).add_columns(B.column_c.label('new_column')).where(A.id == B.class_A_id)

      result = await session.execute(stmt)

       my_result = result.mapping()

but it return `{'A': <.app.models.A object at 0x000001C2F6A37248>, 'new_column': 'value'}`

My expect return is like `{'id': classA_id, column_a: 'value', column_b: 'value', 'new_column': 'value'}` or model A object include new_column

**Comment by CaselIT** [comment]
Hi,

You are using a session and querying an orm model, so the session will return instances of the orm class.
If you want to just get the columns you can either:
- use a connection (using `session.connection().execute(...)`)
- query the A table (using `select(A.__table__)`)
- specify the A column directly (using `select(A.id, A.column_a, A.column_b)`

**Reply by Jasminecll** [reply]
> Hi,
> 
> You are using a session and querying an orm model, so the session will return instances of the orm class. If you want to just get the columns you can either:
> 
> * use a connection (using `session.connection().execute(...)`)
> * query the A table (using `select(A.__table__)`)
> * specify the A column directly (using `select(A.id, A.column_a, A.column_b)`

sure, i mean, my expect result is: 

`{'id': classA_id.value, column_a: 'classA_column_a.value', column_b: 'classA_column_b.value', 'new_column': 'classB_column_c.value'}`
 
I should use `select(A.__table__)` and convert ORM model class to dict, like:

 `{'id': classA_id, column_a: 'value', column_b: 'value'}`. 

Then, I use `tableA.id` to get target `tableB.column_c`, like this:

`select(B.column_c).where(B.class_A_id == tableA.id)`

After that, I can only add a key-value, like `new_column:  tableB.column_c` to the dict to get my expect result.

I did twice query with it, is there has anyway to get my expect result with only once query?

**Reply by Jasminecll** [reply]
> Hi,
> 
> You are using a session and querying an orm model, so the session will return instances of the orm class. If you want to just get the columns you can either:
> 
> * use a connection (using `session.connection().execute(...)`)
> * query the A table (using `select(A.__table__)`)
> * specify the A column directly (using `select(A.id, A.column_a, A.column_b)`



> Hi,
> 
> You are using a session and querying an orm model, so the session will return instances of the orm class. If you want to just get the columns you can either:
> 
> * use a connection (using `session.connection().execute(...)`)
> * query the A table (using `select(A.__table__)`)
> * specify the A column directly (using `select(A.id, A.column_a, A.column_b)`

It did twice query everytimes, So if I want to get a list of that result, It will bring ` N+1 selects problem`

**Reply by CaselIT** [reply]
you would to handle that manually, deduplicating the result as appropriate.
the alternative is to use the ORM with the proper [eager load](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html?highlight=loading) set and then transform that to dicts
